Mage.Sets/src/mage/cards/r/RoryWilliams.java

```diff
@@ -0,0 +1,150 @@
+package mage.cards.r;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.MageObject;
+import mage.MageObjectReference;
+import mage.abilities.Ability;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.continuous.GainSuspendEffect;
+import mage.abilities.effects.keyword.InvestigateEffect;
+import mage.abilities.keyword.*;
+import mage.cards.Card;
+import mage.constants.*;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.counters.CounterType;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.stack.Spell;
+import mage.players.Player;
+
+
+/**
+ *
+ * @author Skiwkr
+ */
+public final class RoryWilliams extends CardImpl {
+
+    public RoryWilliams(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{W}{U}");
+        
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.SOLDIER);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // Partner with Amy Pond
+        this.addAbility(new PartnerWithAbility("Amy Pond"));
+
+        // First strike
+        this.addAbility(FirstStrikeAbility.getInstance());
+
+        // Lifelink
+        this.addAbility(LifelinkAbility.getInstance());
+
+        // The Last Centurion -- When you cast this spell from anywhere other than exile, exile it with three time counters on it. It gains suspend. Then investigate.
+        Ability ability = new RoryWilliamsTriggeredAbility(new RoryWilliamsEffect());
+        ability.addEffect(new InvestigateEffect(1).concatBy(", then"));
+        this.addAbility(ability);
+    }
+
+    private RoryWilliams(final RoryWilliams card) {
+        super(card);
+    }
+
+    @Override
+    public RoryWilliams copy() {
+        return new RoryWilliams(this);
+    }
+}
+
+class RoryWilliamsTriggeredAbility extends TriggeredAbilityImpl {
+
+    public static final String SOURCE_CAST_SPELL_ABILITY = "sourceCastSpellAbility";
+
+    public RoryWilliamsTriggeredAbility(Effect effect) {
+        this(effect, false);
+    }
+
+    public RoryWilliamsTriggeredAbility(Effect effect, boolean optional) {
+        super(Zone.STACK, effect, optional);
+        this.ruleAtTheTop = true;
```

Comment:
(unrelated to this PR, but this should be refactored to use the setter method rather than the field)